src/domain/kaivuilmoitus/Areas.tsx

```diff
@@ -289,8 +289,18 @@ export default function Areas({ hankeData, hankkeenHakemukset, originalHakemus }
       // If the new tyoalue is contained in exactly one hanke area, add it to that
       addTyoAlueToHankeArea(hankeAlueetContainingNewArea[0], feature);
     } else {
-      // New työalue is contained in multiple hanke areas, open dialog for user to select one
-      setMultipleHankeAreaSpanningFeature(feature);
+      // check is the new tyoalue inside more than one hanke area
+      const allAreasContainFeature = hankeAlueetContainingNewArea.every((alue) => {
+        const hankeAlueFeature = alue.geometriat?.featureCollection.features[0];
+        return hankeAlueFeature && featureContains(hankeAlueFeature, newAreaPolygon);
+      });
```

Comment:
Just above on lines 279-282, we filter `hankeAlueetContainingNewArea` to contain the elements that match this test. Checking that the elements still pass this test feels redundant.